core/data/src/main/java/com/yapp/core/data/remote/retrofit/TokenAuthenticator.kt

```
@@ -0,0 +1,56 @@
+            val request = ReissueTokenRequest(
+                accessToken = accessToken,
+                refreshToken = refreshToken
+            )
+
+            val (newAccessToken, newRefreshToken) = try {
+                val response = authApi.reissueToken(request)
+                securityPreferences.setAccessToken(response.accessToken)
+                securityPreferences.setRefreshToken(response.refreshToken)
+
+                response.accessToken to response.refreshToken
+            } catch (e: Exception) {
```

detekt (1.23.7)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)